tests/Polykey.unit.test.ts

```diff
@@ -0,0 +1,155 @@
+// ts imports
+import fs from 'fs'
+import os from 'os'
+import Polykey from '../src/Polykey'
+
+// js imports
+const kbpgp = require('kbpgp')
+const vfs = require('virtualfs')
+
+function randomString(): string {
+	return Math.random().toString(36).replace(/[^a-z]+/g, '').substr(0, 5)
+}
+
+describe('PolyKey class', () => {
+
+	let tempDir: string
+    let pk: Polykey
+
+	beforeAll(done => {
+		// Define temp directory
+		tempDir = fs.mkdtempSync(`${os.tmpdir}/pktest${randomString()}`)
+
+		// Create keyManager
+		const km = new Polykey.KeyManager()
+		km.loadKeyPair('./playground/keys/private.key', './playground/keys/public.key')
+		// Initialize polykey
+		pk = new Polykey(
+			Buffer.from('some passphrase'),
+			km,
+			32,
+			tempDir
+		)
+		done()
+	})
+
+	afterAll(() => {
+		fs.rmdirSync(`${tempDir}`)
+	})
+
+	////////////
+	// Vaults //
+	////////////
+	describe('vaults', () => {
```

Comment:
I think `vault` tests should be in a separate file entirely.

You want to test a couple things. Start with the domain models first. And then we wrap it like an onion and then testing larger things, to eventually the entire CLI commands.

Reply:
Vault tests have been moved to a separate file